sdk_extensions/aws_v1_support/src/main/java/io/opentelemetry/sdk/extensions/trace/aws/resource/JdkHttpClient.java

```diff
@@ -0,0 +1,148 @@
+        if (keyStore != null) {
+          ((HttpsURLConnection) connection).setSSLSocketFactory(buildSslSocketFactory(keyStore));
+        }
+      } else {
+        connection = (HttpURLConnection) new URL(urlStr).openConnection();
+      }
+
+      connection.setRequestMethod(httpMethod);
+      connection.setConnectTimeout(TIMEOUT_MILLIS);
+      connection.setReadTimeout(TIMEOUT_MILLIS);
+
+      for (Map.Entry<String, String> requestProperty : requestPropertyMap.entrySet()) {
+        connection.setRequestProperty(requestProperty.getKey(), requestProperty.getValue());
+      }
+
+      int responseCode = connection.getResponseCode();
+      if (responseCode != 200) {
+        logger.log(
+            Level.FINE,
+            "Error reponse from "
+                + urlStr
+                + " code ("
+                + responseCode
+                + ") text "
+                + readResponseString(connection));
+        return "";
+      }
+
+      return readResponseString(connection).trim();
+
+    } catch (IOException e) {
+      logger.log(Level.FINE, "JdkHttpClient fetch string failed.", e);
+    }
+
+    return "";
+  }
+
+  private static String readResponseString(HttpURLConnection connection) {
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+    try (InputStream is = connection.getInputStream()) {
+      ByteStreams.copy(is, os);
+    } catch (IOException e) {
+      // Only best effort read if we can.
+    }
+    try (InputStream is = connection.getErrorStream()) {
+      if (is != null) {
+        ByteStreams.copy(is, os);
+      }
+    } catch (IOException e) {
+      // Only best effort read if we can.
+    }
+    try {
+      return os.toString(StandardCharsets.UTF_8.name());
+    } catch (UnsupportedEncodingException e) {
+      logger.log(Level.WARNING, "UTF-8 not supported can't happen.", e);
+    }
+    return "";
+  }
+
+  private static SSLSocketFactory buildSslSocketFactory(KeyStore keyStore) {
+    try {
+      String tmfAlgorithm = TrustManagerFactory.getDefaultAlgorithm();
+      TrustManagerFactory tmf = TrustManagerFactory.getInstance(tmfAlgorithm);
+      tmf.init(keyStore);
+
+      SSLContext context = SSLContext.getInstance("TLS");
+      context.init(null, tmf.getTrustManagers(), null);
+      return context.getSocketFactory();
+
+    } catch (Exception e) {
+      logger.log(Level.WARNING, "Build SslSocketFactory for K8s restful client exception.", e);
+    }
+    return null;
+  }
+
+  private static KeyStore getKeystoreForTrustedCert(String certPath) {
+    try (FileInputStream fis = new FileInputStream(certPath)) {
+      KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
+      trustStore.load(null, null);
+      CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+
+      Collection<? extends Certificate> certificates = certificateFactory.generateCertificates(fis);
+
+      int i = 0;
+      for (Certificate certificate : certificates) {
```

Comment:
Ah good catch - if we need index let's just use traditional for loop. `for(int i = 0; I < ...`

Reply:
Collection does not support random access, have to convert it to Array. Here I also fell code looks ugly, if we have other way to name the cert in for loop, I would prefer not call Collection.toArray().

Reply:
Ah was thinking collection has it as opposed to iterable. In that case no worries.